central/cve/converter/utils/convert_utils_test.go

```diff
@@ -1,242 +1,51 @@
-			Cvss:         float32(cves[1].Impact.BaseMetricV3.CVSSV3.BaseScore),
-			Summary:      "Description3",
-			Link:         "https://nvd.nist.gov/vuln/detail/cve-2019-2",
-			ScoreVersion: storage.CVE_V3,
-			CvssV2: &storage.CVSSV2{
-				Vector:              "AV:N/AC:L/Au:S/C:N/I:P/A:N",
-				AttackVector:        storage.CVSSV2_ATTACK_NETWORK,
-				AccessComplexity:    storage.CVSSV2_ACCESS_LOW,
-				Authentication:      storage.CVSSV2_AUTH_SINGLE,
-				Confidentiality:     storage.CVSSV2_IMPACT_NONE,
-				Integrity:           storage.CVSSV2_IMPACT_PARTIAL,
-				Availability:        storage.CVSSV2_IMPACT_NONE,
-				ExploitabilityScore: float32(cves[1].Impact.BaseMetricV2.ExploitabilityScore),
-				ImpactScore:         float32(cves[1].Impact.BaseMetricV2.ImpactScore),
-				Score:               float32(cves[1].Impact.BaseMetricV2.CVSSV2.BaseScore),
-				Severity:            storage.CVSSV2_MEDIUM,
-			},
-			CvssV3: &storage.CVSSV3{
-				Vector:              "CVSS:3.0/AV:N/AC:L/PR:L/UI:N/S:C/C:N/I:H/A:N",
-				ExploitabilityScore: float32(cves[1].Impact.BaseMetricV3.ExploitabilityScore),
-				ImpactScore:         float32(cves[1].Impact.BaseMetricV3.ImpactScore),
-				AttackVector:        storage.CVSSV3_ATTACK_NETWORK,
-				AttackComplexity:    storage.CVSSV3_COMPLEXITY_LOW,
-				PrivilegesRequired:  storage.CVSSV3_PRIVILEGE_LOW,
-				UserInteraction:     storage.CVSSV3_UI_NONE,
-				Scope:               storage.CVSSV3_CHANGED,
-				Confidentiality:     storage.CVSSV3_IMPACT_NONE,
-				Integrity:           storage.CVSSV3_IMPACT_HIGH,
-				Availability:        storage.CVSSV3_IMPACT_NONE,
-				Score:               float32(cves[1].Impact.BaseMetricV3.CVSSV3.BaseScore),
-				Severity:            storage.CVSSV3_HIGH,
-			},
-			Type: storage.CVE_K8S_CVE,
-		},
-	}
-
-	for i := 0; i < len(cves); i++ {
-		expectedVul := &expectedVuls[i]
+func TestNVDCVEToEmbeddedCVEs(t *testing.T) {
+	data, err := os.ReadFile("testdata/cve-list.json")
+	require.NoError(t, err)
+	var cveEntries []*schema.NVDCVEFeedJSON10DefCVEItem
+	require.NoError(t, json.Unmarshal(data, &cveEntries))
 
-		ts, err := time.Parse(timeFormat, publishedDateTime)
-		assert.Nil(t, err)
-		expectedVul.PublishedOn = protoconv.ConvertTimeToTimestamp(ts)
-
-		ts, err = time.Parse(timeFormat, lastModifiedDateTime)
-		assert.Nil(t, err)
-		expectedVul.LastModified = protoconv.ConvertTimeToTimestamp(ts)
+	embeddedCVEs, err := NVDCVEsToEmbeddedCVEs(cveEntries, Istio)
+	assert.NoError(t, err)
+	assert.Len(t, embeddedCVEs, len(cveEntries))
+}
 
-		actualVul, err := NvdCVEToProtoCVE(cves[i], K8s)
-		assert.Nil(t, err)
-		assert.Equal(t, actualVul, expectedVul)
+func TestNVDCVEToEmbeddedCVE(t *testing.T) {
+	data, err := os.ReadFile("testdata/cve-list.json")
+	require.NoError(t, err)
+	var cveEntries []*schema.NVDCVEFeedJSON10DefCVEItem
+	require.NoError(t, json.Unmarshal(data, &cveEntries))
+
+	for _, cveEntry := range cveEntries {
+		t.Run(cveEntry.CVE.CVEDataMeta.ID, func(t *testing.T) {
```

Comment:
```suggestion
		cveEntry := cveEntry
		t.Run(cveEntry.CVE.CVEDataMeta.ID, func(t *testing.T) {
```
to de-alias

Reply:
Good catch, I forgot this is a slice of pointers